### PR TITLE
Migrate to github actions

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Migrate to GitHub actions from Travis-CI.

TODO: some repo admin work needs to be done by a folk with required permissions: `continuous-integration/travis-ci Expected — Waiting for status to be reported`.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
All the other flyte components are on GitHub actions already, so better be this one, plus travis-ci.org is being shut down.

## Tracking Issue
_NA_

## Follow-up issue
Re-purpose this issue: https://github.com/lyft/flyte/issues/669

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -4,27 +4,101 @@
...
-pycodestyle==2.6.0        # via flake8
-pyflakes==2.2.0           # via flake8
-pyparsing==2.4.7          # via packaging
-pytest==6.1.2             # via -r dev-requirements.in
-regex==2020.11.13         # via -c requirements.txt, black
-testfixtures==6.15.0      # via flake8-isort
-toml==0.10.2              # via -c requirements.txt, black, pytest
-typed-ast==1.4.1          # via -c requirements.txt, black
+appdirs==1.4.4
+    # via
```
Comment (honnix) [inline dev-requirements.txt:8]: These change are caused by https://github.com/jazzband/pip-tools/pull/1237

File: dev-requirements.in [hunk 2]
```diff
@@ -8,3 +8,4 @@ flake8-isort
 isort
 mock
 pytest
+codecov
```
Comment (honnix) [inline dev-requirements.in:11]: This has been missing for a while.

**Comment by honnix** [comment]
@wild-endeavor @kumare3 PTAL, thanks.

File: .github/workflows/pythonbuild.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+        - python-version: 3.8
```
Comment (honnix) [inline .github/workflows/pythonbuild.yml:15]: I didn't bother including 3.6 as it is dying away very soon as I understood.

**Comment by honnix** [comment]
Wow! `Coverage   81.00%   96.00%   +14.99%`

**Comment by kumare3** [comment]
How did you increase the coverage?

**Comment by honnix** [comment]
> How did you increase the coverage?

@kumare3 No I didn't. I guess it's just we have not been uploading coverage report for some time, since #154 which deleted `codecov` installation.

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
I don't have any admin permissions to this repo anymore though.  Maybe @anandswaminathan does?

**Comment by wild-endeavor** [comment]
Trying to get some Lyft internal people to help out...

**Review by anandswaminathan — APPROVED** [review]

**Review by bnsblue — APPROVED** [review]

**Comment by honnix** [comment]
@anandswaminathan or @bnsblue could you please remove the required CI and change it to GitHub actions? Otherwise I cannot merge. Thanks!
